sdk/src/lib.rs

```diff
@@ -85,68 +86,92 @@ impl ProverClient {
     pub fn prove(&self, elf: &[u8], stdin: SP1Stdin) -> Result<SP1ProofWithIO<BabyBearPoseidon2>> {
         if self.client.is_some() {
             println!("Proving remotely");
-            self.prove_remote(elf, stdin)
+            self.prove_remote_sync(elf, stdin)
         } else {
             println!("Proving locally");
             self.prove_local(elf, stdin, BabyBearPoseidon2::new())
         }
     }
 
-    pub fn prove_remote(
+    /// Generate a proof for the execution of the ELF with the given public inputs for an async context
+    pub async fn prove_async(
```

Comment:
kind of weird to have *_async and *_sync functions - I think we should name the async version of `prove_remote` to `prove_remote_async`, and leave the sync version as just `prove_remote`.

Reply:
Updated!